src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -825,7 +863,8 @@ private InputsAndSuccessFlag selectInputs(TypedOperation operation) {
         }
       }
 
-      variables.add(totStatements + randomVariable.index);
+      variables.add(
```

Comment:
TODO: figure out what variables does (line 696 of this file)

Reply:
Looks like it represents the indices to the position of input variables for methods under test. The sequence `S` is represented as a list of Sequences (concatenated at the end), of which each Sequence is a List of Statements used to construct the input.